### DISCUSSION TITLE
Planning next release

### DISCUSSION THREAD

**Discussion by dreua** [pdfarranger/pdfarranger#1077] [General]
Do we have any plans for the next release? I'd need to decide if I patch the flathub build downstream with the appstream data updates in order to rebuild it with updated libraries but I'd rather not. (Mainly for reasons of stubbornness and laziness tbh. Still a bit annoyed by the flathub team just breaking stuff and us maintainers needing to pick up the pieces.)

In general it might be a good idea because we have some new features and at least one bug fixed that is not yet released. I'd suggest to wait for #1065 and #1066 for another great feature addition :rocket: 

Maybe after they are reviewed and merged we call for translations and then do a release? Any thoughts?

Edit: Reminder to update the pot/po files before call for translations and the metainfo before release.

**Comment by kbengs** [comment]
I think it would be a good time for a release. With or without 1065 and 1066 is fine imo.
I added some PR:s to 1.11 milesone https://github.com/pdfarranger/pdfarranger/pulls?q=is%3Apr+milestone%3A1.11+is%3Aclosed
But maybe version 2.0 would be better as suggested earlier?

Having a checklist for the release would be handy. But I am happy if Jerome handles it as usual.

**Reply by dreua** [reply]
Did we have breaking changes or changes in the dependencies? That would be arguments for 2.0 but I don't really remember, maybe I missed something. Both 2.0 and 1.11 is fine with me, no need to discuss it in depth imo.

Fully agree in all other points.

**Reply by kbengs** [reply]
We had at least https://github.com/pdfarranger/pdfarranger/pull/1014

**Comment by bagajohny** [comment]
Any update on the next release? When will it be available?

**Reply by dreua** [reply]
We don't have fixed schedules around here, in the next 1-2 month maybe ;)

**Comment by kbengs** [comment]
Proposed checklist for releasing new version

1. Hint translators
2. Add PR/issues/commits to milestone
3. Draft a new release
4. Create commit: "Bump version to 1.11.0"
   See 1.10.1 Commit: https://github.com/pdfarranger/pdfarranger/commit/1d2d46deed957686cd527ef394fc231d0e140db8
5. Add new tag on "create a new release" page
6. Test zip and msi
7. Add Assets:
   pdfarranger-1.11.0-windows-installer.msi (extracted msi copied from from Action page)
   pdfarranger-1.11.0-windows-portable.zip (extracted zip copied from from Action page)
   (Source code is added automatically)
8. Publish release

We already got several translations. 2-3 is done. As we got the pikepdf 9 issues, if no one argue I can go ahead with the rest of the steps.

**Reply by dreua** [reply]
Re 7: At least the source code archives will be added by github automatically, I'm not sure if that also applies to the msi/zip though :thinking:

**Reply by kbengs** [reply]
1.11.0 is released :smile: added zip and msi manually.

**Comment by kbengs** [comment]
Should we release 1.13.0, what do you  think @dreua ? The release notes is drafted.

**Reply by dreua** [reply]
Looks good to me, thanks for the draft!

In the past we mentioned translations updates and new translations in the release notes, not sure if we want to keep that up. (I remember merging at least two new translations leading up to a total of 33 translations iirc :partying_face: )

I would have liked to get rid of dateutil #1268 but that should not be a holdup now, hopefully I'll find some time to take care of it after the release so we have a good testing period.

Should there be a call for translations (did we add or change any strings?) and some time for testing?

**Reply by kbengs** [reply]
Yeah we could mention the two new translations. I added that. But maybe not necessary to mention updated translations.

I think pdfarranger.pot is updated except for "Software versions:". Do you want to call for translators? And maybe release in 3-4 weeks? I'm also fine with leaving that step out and maybe release this weekend.

**Reply by dreua** [reply]
Yeah let's not over complicate things, if you want to release this weekend that is great, go ahead :rocket: :)

**Comment by kbengs** [comment]
I drafted 1.14.0 release notes. Should we do a release soon?

**Reply by dreua** [reply]
Fine by me, I recently stumbled over the broken language list myself so that would be a good reason to get a fixed version out to the users for me.
Sooner or later I'd like to see #1268 and #1265 fixed or at least looked at and decided but it is still not a blocker and I guess I can't make anyone do that and just have to do it myself someday :laughing: 

Thank you for all the work you put into that, most recently the windows built environment (that looked like a bit of work, only seeing the results...)!

Should we put the upated openssl lib in the release notes? On the one hand we don't know whether it was exploitable at all in PDF Arranger (I think rather unlikely) but on the other hand those users who had alerts in their BOM scanners will be glad to see it.

**Reply by dreua** [reply]
(BOM = bill of materials)
